### PR TITLE
Add an admin section, with auth requiring an admin

### PR DESCRIPTION
**Pull request by mileszs**
I have some work to do here, yet, but I wanted to get this out there so it was clear what is being worked on. 

This creates an admin section, `/admin/jobs` and the related endpoints. I've not actually done anything with delete yet. Nor have I written tests for the plug I created, or the admin controllers, etc.

### CODE REVIEW

**Comment by mileszs** [comment]
Alright, I think this is complete now.

File: test/controllers/admin/job_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,103 @@
+defmodule ElixirJobBoard.JobControllerTest do
+  use ElixirJobBoard.ConnCase
+  alias ElixirJobBoard.Job
+  alias ElixirJobBoard.User
+  import ElixirJobBoard.Factory
+
+  setup do
+    user = ElixirJobBoard.Factory.insert(:user, %{ admin: true })
```
Comment (andrewariley87) [inline test/controllers/admin/job_controller_test.exs]: I believe this line can be shortened to `user = insert(:user, admin: true)`.

Comment (andrewariley87) [inline test/controllers/admin/job_controller_test.exs]: I was also writing tests for this controller and I found this test in my Phoenix Programming book.

```
test "all actions require user to be a admin",  %{conn: conn} do
    user = insert(:user)
    job = insert(:job)
    conn = assign(build_conn(), :current_user, user)
    {:ok, conn: conn, user: user}
    refute User.is_admin?(user)

    Enum.each([
      get(conn, job_path(conn, :new)),
      get(conn, job_path(conn, :index)),
      get(conn, job_path(conn, :show, job)),
      get(conn, job_path(conn, :edit, job)),
      post(conn, job_path(conn, :create, %{})),
      put(conn, job_path(conn, :update, job, %{}))
      ], fn conn ->
        assert html_response(conn, 302)
        assert conn.halted
    end)
  end
```

I don't know if it is necessary but thought it was a cool way to show that unless a user was an admin they do not have any access to the actions in this controller.

**Comment by andrewariley87** [comment]
👍  with comments.

File: test/controllers/admin/job_controller_test.exs (hunk 1 above)
Comment (mileszs) [reply test/controllers/admin/job_controller_test.exs]: Fixed

Comment (mileszs) [reply test/controllers/admin/job_controller_test.exs]: Nice! Added.

**Pull request merged**